### DISCUSSION TITLE
Failed to load the submitted data due to invalid formatting - HTTP Request POST

### DISCUSSION THREAD

**Discussion by matheusoreis** [pocketbase/pocketbase#3009] [Q&A]
I'm developing an app in Flutter, but I'm not using the SDK, I'm using standard HTTP requests.

But when I try to send a POST request to register a user or a "test" according to the print that I will leave here, it returns the following error:

`
{
    "code": 400,
    "message": "Failed to load the submitted data due to invalid formatting.",
    "data": {}
}
`

The code for sending the request looks like this:
`
const String url = 'http://192.168.1.103:8090/api/collections/teste/records';

    final response = await iHttpClient.post(
      url: url,
      body: <String, dynamic>{
        "teste": "samuel@gmail.com",
      },
    );

    if (response.statusCode == 200) {
      // Decodifica o Body JSON para um Map<String, dynamic>
      final Map<String, dynamic> body = jsonDecode(response.body);

      final signUpResponseModel = SignUpData.fromJson(body);
      return (null, signUpResponseModel);
    } else {
      final Map<String, dynamic> body = jsonDecode(response.body);

      final signUpResponseModel = ErrorResponse.fromJson(body);

      return (signUpResponseModel, null);
    }`

![image](https://github.com/pocketbase/pocketbase/assets/47448206/543acbb6-92d8-4b73-8b1a-a03cf011fbf5)

What I'm finding funnier is that when I try to do the same thing but for password authentication, it works perfectly but when it's a POST to create the same it returns this to me.

``
class SignUpRepository {
  final IHttpClient iHttpClient;

  SignUpRepository({required this.iHttpClient});

  Future<Result<ErrorResponse, SignUpData>> signUp() async {
    const String url = "http://192.168.1.103:8090/api/collections/teste/records";

    final response = await iHttpClient.post(
      url: url,
      body: <String, dynamic>{
        "teste": "samuel@gmail.com",
      },
    );

    print(response.body);

    if (response.statusCode == 200) {
      // Decodifica o Body JSON para um Map<String, dynamic>
      final Map<String, dynamic> body = jsonDecode(response.body);

      final signUpResponseModel = SignUpData.fromJson(body);
      return (null, signUpResponseModel);
    } else {
      final Map<String, dynamic> body = jsonDecode(response.body);

      final signUpResponseModel = ErrorResponse.fromJson(body);

      return (signUpResponseModel, null);
    }
  }
}
``

**Comment by ganigeorgiev** [comment]
If you are sending a JSON make sure that your request has the `Content-Type: application/json` header.

**Comment by matheusoreis** [comment]
Thanks a lot, I forced the content-type and it worked! I hadn't noticed that Pocketbase has such a powerful logging system.

Thank you so much for your support and for this great software!

Brazil loves you!
